src/HS/ListingBundle/Controller/DefaultController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace HS\ListingBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Doctrine\ORM\EntityManager;
+use HS\ListingBundle\Repository\ListingRepository;
+use HS\ListingBundle\Entity\Listing;
+use HS\ListingBundle\Form\ListingType;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use OC\PlatformBundle\Event\MessagePostEvent;
+use OC\PlatformBundle\Event\PlatformEvents;
+
+
+
+class DefaultController extends Controller
+{
+    
+    /**
+     * 
+     * @Security("has_role('ROLE_USER')")
+     * @Route("/manage/listings", name="hs_listing_index")
+     */
+    public function indexAction()
+    {
+        //get entityManager 
+        $listingRepository = $this->getDoctrine()->getManager()
+            ->getRepository(Listing::class);
+
+        //current connected user
+    	$user = $this->getUser();
+
+        //get listings from repository    
+        $listings = $listingRepository->getUserListings($user->getId());
+
+        //render view and send $listings as argument
+        return $this->render('HSListingBundle:Listing:index.html.twig', array(
+            'listings' => $listings
+        ));
+    }
+
+    /**
+     * 
+     * @Security("has_role('ROLE_USER')")
+     * @Route("/manage/listing", name="hs_listing_add")
+     */
+    public function addAction(Request $request)
+    {
+        $listing = new Listing();
```

Comment:
@Reddaa 

- [ ] Controller is too big, you need to use Services to externalize business code


- [ ] No Exception handling

Side Note : prkoi t'as mis le meme controlleur et pour le POST et pour le GET ?

Reply:
Sorry @ahaizoune  I had some problems with git's auto merging tool. 
The methods for get and POST has been separated, although it's a kind of repetition with the new code.
Exception handling: I can't see any exception that will be thrown, can you provide an example please 👍 
I created a service named file_mover to move the uploaded files to the public directory.

I'll push changes to a new branch. this morning's problem had me scared 🔨

Reply:
@Reddaa 
- You can implement two controllers & put the code inside a Service Class. That way, the 2 controllers use the same Service to avoid code replication.
- The simplest example : if the database crashs, you need to catch the error in the runtime & return an HTTP code 500 

poke @coffeemakesmelovelife